### DISCUSSION TITLE
Use `next-iron-session` in `with-passport` example

### DISCUSSION THREAD

**Discussion by Kerumen** [vercel/next.js#12387] [Help]
I'm wondering if it's a good idea to merge the cookie handling of the example [with-iron-session](https://github.com/zeit/next.js/tree/canary/examples/with-iron-session) to the [with-passport](https://github.com/zeit/next.js/tree/canary/examples/with-passport) example.

For example, there are extraneous work done in the passport example ([here](https://github.com/zeit/next.js/blob/canary/examples/with-passport/pages/api/login.js#L27-L30)) that is already covered by the library `next-iron-session`.

_I'm not talking about merging both examples, but only to use `next-iron-session` in the `with-passport` example._

As a side question, I was wondering why the example is not using passport's session implementation. There are some examples out there (https://github.com/hoangvvo/nextjs-mongodb-app) and I can't tell why the `with-passport` example is using a custom cookie. If there are an auth expert passing by, I'd love to know!

Pinging the iron-session author (@vvo), the passport example author (@lfades) and @hoangvvo.

Thanks!

**Comment by vvo** [comment]
I just published updates to next-iron-session that may or may not ease the with-passport example: https://github.com/vvo/next-iron-session/releases

You now have `applySession(req, res, options)` and `ironSession(options)` (connect middleware).

**Comment by lfades** [comment]
If the `with-passport` example can be updated to implement some better behavior from `with-iron-session` then please create a PR. I don't think we have to merge them as the `with-passport` example shows how to setup a secure cookie and a good authentication token by yourself, and `next-iron-session` does basically the same.

On the passport note, I chose not to go very deep into the Passport API as I don't really like it and it feels very old.

I'm not an auth expert, but my best recommendation if you're looking for security is to take a look into the auth examples with third party services, we have examples for Userbase, Auth0 and Magic Link. Making a very secure example would take way more code and validations, and it may not look like an example anymore. Authentication is hard.
